### PR TITLE
LibWeb: Implement HTML DragEvent class

### PR DESCRIPTION
**Pull request by sppmacd**
This just defines the class, drag events aren't actually fired yet.

Progress on https://jupyter.org/try-jupyter/lab/ (requires #498):
image

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/DragEvent.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    return realm.heap().allocate<DragEvent>(realm, realm, event_name, event_init);
+}
+
+WebIDL::ExceptionOr<JS::NonnullGCPtr<DragEvent>> DragEvent::construct_impl(JS::Realm& realm, FlyString const& event_name, DragEventInit const& event_init)
+{
+    return create(realm, event_name, event_init);
+}
+
+DragEvent::DragEvent(JS::Realm& realm, FlyString const& event_name, DragEventInit const& event_init)
+    // FIXME: What are the correct values for page_x, page_y, offset_x, offset_y?
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/DragEvent.cpp:26]: For mouse events these values are passed to `MouseEvent::create()`, but are defaulted to 0. Is that something we could do here?

FYI, for `MouseEvent` these values are initially provided by `EventHandler`:

https://github.com/LadybirdBrowser/ladybird/blob/31c7e98a4a46c2d0ef93c5fca47d64d05b96449f/Userland/Libraries/LibWeb/Page/EventHandler.cpp#L272-L275

Comment (sppmacd) [reply Userland/Libraries/LibWeb/HTML/DragEvent.cpp]: Moved these params up to the constructor / `DragEvent::create`.

File: Userland/Libraries/LibWeb/HTML/DragEvent.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Copyright (c) 2024, Maciej <sppmacd@pm.me>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/DragEventPrototype.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/HTML/DragEvent.h>
+
+namespace Web::HTML {
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/DragEvent.cpp:11]: Other event classes like this are in the `Web::UIEvents` namespace, does this belong there too?

Comment (sppmacd) [reply Userland/Libraries/LibWeb/HTML/DragEvent.cpp:11]: It's defined in the HTML spec: https://html.spec.whatwg.org/multipage/dnd.html#the-dragevent-interface

**Review by tcl3 — APPROVED** [review]

**Pull request merged**